test/core/event_engine/posix/event_poller_posix_test.cc

```diff
@@ -376,14 +380,8 @@ void WaitAndShutdown(server* sv, client* cl) {
   gpr_mu_lock(&g_mu);
   while (!sv->done || !cl->done) {
     gpr_mu_unlock(&g_mu);
-    result = g_event_poller->Work(24h);
-    if (absl::holds_alternative<Poller::Events>(result)) {
-      auto pending_events = absl::get<Poller::Events>(result);
-      for (auto it = pending_events.begin(); it != pending_events.end(); ++it) {
-        (*it)->Run();
-      }
-      pending_events.clear();
-    }
+    result = g_event_poller->Work(24h, []() {});
```

Comment:
Maybe make these callbacks assert called/notcalled?

Reply:
I think that could make the test flaky since there is a possibility that the test can finish in one iteration and sometimes it might require more than one iteration of Work